### PR TITLE
feat(plugin): Changed MongoDB Logs plugin to parse to body

### PR DESCRIPTION
**Pull request by cpheps**
### Proposed Change
- Removed unused parameter `retain_raw_logs` from tomcat plugin
- Added parsing to body to MongoDB logs plugin. 

MongoDB older style logs with `retain_raw_logs` enabled:

```
LogRecord #1
ObservedTimestamp: 2022-09-08 17:37:23.165769 +0000 UTC
Timestamp: 2019-02-06 14:22:27.347 +0000 UTC
Severity: I
Body: {
     -> component: STRING(CONTROL)
     -> context: STRING(initandlisten)
     -> message: STRING(db version v3.6.2)
     -> raw_log: STRING(2019-02-06T09:22:27.347-0500 I CONTROL  [initandlisten] db version v3.6.2)
     -> severity: STRING(I)
     -> timestamp: STRING(2019-02-06T09:22:27.347-0500)
}
Attributes:
     -> log_type: STRING(mongodb)
     -> log.file.name: STRING(old.log)
Trace ID: 
Span ID: 
Flags: 0
```

MongoDB newer json style logs with `retain_raw_logs` enabled:

```
LogRecord #1
ObservedTimestamp: 2022-09-08 18:01:17.044687 +0000 UTC
Timestamp: 2020-11-03 19:24:05.178 +0000 UTC
Severity: W
Body: {
     -> component: STRING(ASIO)
     -> context: STRING(main)
     -> id: DOUBLE(22601)
     -> message: STRING(No TransportLayer configured during NetworkInterface startup)
     -> raw_log: STRING({\"t\":{\"$date\":\"2020-11-03T14:24:05.178-05:00\"},\"s\":\"W\",  \"c\":\"ASIO\",     \"id\":22601,   \"ctx\":\"main\",\"msg\":\"No TransportLayer configured during NetworkInterface startup\"})
     -> s: STRING(W)
     -> t: MAP({\"$date\":\"2020-11-03T14:24:05.178-05:00\"})
}
Attributes:
     -> log.file.name: STRING(new.log)
     -> log_type: STRING(mongodb)
Trace ID: 
Span ID: 
Flags: 0
```

##### Checklist
- [x] Changes are tested
- [ ] CI has passed

### CODE REVIEW

**Comment by StefanKurek** [comment]
@cpheps in your summary example, is it expected that there is no timestamp for the second log type?

**Comment by BinaryFissionGames** [comment]
Looks like the newer logs aren't correctly parsing the timestamp; Was this always this way or was it introduced w/ this PR?

**Comment by cpheps** [comment]
Good catch on the timestamps let me double check that

File: plugins/mongodb_logs.yaml [hunk 1]
```diff
@@ -61,13 +72,13 @@ template: |
         # {"t":{"$date":"2022-04-26T16:15:44.876+00:00"},"s":"I",  "c":"INDEX",    "id":20345,   "ctx":"LogicalSessionCacheRefresh","msg":"Index build: done building","attr":{"buildUUID":null,"namespace":"config.system.sessions","index":"lsidTTLIndex","commitTimestamp":null}}
         - id: 4_4_parser
           type: json_parser
-          parse_from: body
+          parse_to: body
           timestamp:
-            parse_from: attributes.t.$date
+            parse_from: body.t.$date
```
Comment (BinaryFissionGames) [inline plugins/mongodb_logs.yaml:77]: Looks like this issue was always present, but I think the env-var replacement is biting us here.
```suggestion
            parse_from: body.t.$$date
```

Comment (schmikei) [reply plugins/mongodb_logs.yaml:77]: I'm confused I jsut ran the plugin and it appeared to parse correctly? 

```
LogRecord #99
ObservedTimestamp: 2022-09-08 17:54:33.026062 +0000 UTC
Timestamp: 2022-09-08 17:53:37.549 +0000 UTC
Severity: I
Body: {
     -> component: STRING(CONTROL)
     -> context: STRING(initandlisten)
     -> message: STRING(db version v4.0.25)
     -> raw_log: STRING(2022-09-08T17:53:37.549+0000 I CONTROL  [initandlisten] db version v4.0.25)
     -> severity: STRING(I)
     -> timestamp: STRING(2022-09-08T17:53:37.549+0000)
}
Attributes:
     -> log.file.name: STRING(test.log)
     -> log_type: STRING(mongodb)
Trace ID:
Span ID:
Flags: 0
```

Comment (cpheps) [reply plugins/mongodb_logs.yaml:77]: Yeah just verified it's always broken I'll work on fixing it.

Comment (cpheps) [reply plugins/mongodb_logs.yaml:77]: The double `$$` works. I'll update and retest.

Comment (schmikei) [reply plugins/mongodb_logs.yaml:77]: Ahh now i see the issue nvm me 🙃

**Review by BinaryFissionGames — APPROVED** [review]

**Review by schmikei — APPROVED** [review]

**Pull request merged**